src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -107,6 +113,39 @@ function coerceRef(current: ?Fiber, element: ReactElement) {
   return mixedRef;
 }
 
+function throwOnInvalidObjectType(returnFiber : Fiber, newChild : Object) {
+  if (returnFiber.type !== 'textarea') {
+    const childrenString = String(newChild);
+    let addendum = '';
+    if (__DEV__) {
+      addendum =
+        ' If you meant to render a collection of children, use an array ' +
+        'instead or wrap the object using createFragment(object) from the ' +
+        'React add-ons.';
+      if (newChild._isReactElement) {
```

Comment:
We don't need this check any more; it's legacy. You can delete it and the test if you want.